packages/data/src/redux-store/metadata/reducer.ts

```diff
@@ -36,17 +40,41 @@ const subKeysIsResolved: Reducer< Record< string, State >, Action > = onSubKey<
 	switch ( action.type ) {
 		case 'START_RESOLUTION':
 		case 'FINISH_RESOLUTION': {
-			const isStarting = action.type === 'START_RESOLUTION';
+			const isResolving = action.type === 'START_RESOLUTION';
 			const nextState = new EquivalentKeyMap( state );
-			nextState.set( action.args, isStarting );
+			nextState.set( action.args, { isResolving } );
+			return nextState;
+		}
+		case 'FAIL_RESOLUTION': {
+			const nextState = new EquivalentKeyMap( state );
+			nextState.set( action.args, {
+				isResolving: false,
+				error: action.error,
+			} );
+
 			return nextState;
 		}
 		case 'START_RESOLUTIONS':
 		case 'FINISH_RESOLUTIONS': {
-			const isStarting = action.type === 'START_RESOLUTIONS';
+			const isResolving = action.type === 'START_RESOLUTIONS';
 			const nextState = new EquivalentKeyMap( state );
 			for ( const resolutionArgs of action.args ) {
-				nextState.set( resolutionArgs, isStarting );
+				nextState.set( resolutionArgs, { isResolving } );
+			}
+			return nextState;
+		}
+		case 'FAIL_RESOLUTIONS': {
+			const nextState = new EquivalentKeyMap( state );
+			for ( const idx in action.args ) {
```

Comment:
Going through an array with `in` is a weird style 🙂 I'd do:
```js
action.args.forEach( ( resolutionArgs, idx ) => {
  nextState.set( resolutionArgs, { isResolving: false, error: action.errors[ idx ] } );
}
```

Reply:
Makes sense! I used the `forEach` but kept the conditional setting of the `error` key.